json-rpc/tests/integration_test.rs

```diff
@@ -858,6 +860,71 @@ fn create_test_cases() -> Vec<Test> {
                 }
             },
         },
+        Test {
+            name: "get_transactions_with_proofs",
+            run: |env: &mut testing::Env| {
+                let resp = env.send("get_metadata", json!([]));
+                assert!(resp.libra_ledger_version > 0);
+                for base_version in 0..resp.libra_ledger_version {
```

Comment:
remove this for loop, be specific about what you want to test, set base_version, and I think we should test base_version + limit < resp.libra_ledger_version

Reply:
I'm testing all the possible queries from version 0 to the libra_ledger_version. That's why I have the for loop. I guess I could replace it with just a test for the base_version = 5 or so, but it would cover less stuff, no?

I could add a test where base_version + limit > resp.libra_ledger_version and check it works as intended and returns less than limit txs.
But as long as base_version < resp.libra_ledger_version, this should still work I think. It's a limit, but if the limit is 400 and there are only 100 txs in the response because the ledger_version is equal to base_version+99, then that would still be a valid response, no?